feedparser/datetimes/rfc822.py

```diff
@@ -13,6 +13,12 @@
     'm': -12, 'y': 12,
     'met': 1, 'mest': 2,
 }
+daynames = ('mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun')
```

Comment:
Elements in `daynames` are already unique and since it moved to the global namespace the `set` still would be better. It is only used for checking if a date part is contained in this collection (line 45) so for `tuple` this would mean unnecessary iteration because it is still sequence. In case of miss this is a scan over whole tuple content - O(n). Sets in Python allow for fast lookups and efficient membership testing using `in` operator - O(1). See [PythonSpeed](https://wiki.python.org/moin/PythonSpeed) as a reference.

I assume that redundant set creation on every call was reducing the performance but once it is moved to the global scope there is a substantial gain of using it instead of list or tuple. Mutability is not a problem here. Also, you can move further and use its immutable counterpart - `frozenset`. I don't think if it can give any time performance advantage here but for sure it is more memory efficient because it does not require excessive overallocation of underlying hash table structure.